Fix up ansible-test sanity checks due to ansible 2.17 release

fosterseth opened the pull request:
##### SUMMARY
<!--- Describe the change, including rationale and design decisions -->
Address the following ansible sanity errors

```
ERROR: Found 7 pylint issue(s) which need to be resolved:
ERROR: plugins/module_utils/controller_api.py:110:15: unidiomatic-typecheck: Use isinstance() rather than type() for a typecheck.
ERROR: plugins/module_utils/controller_api.py:218:23: unidiomatic-typecheck: Use isinstance() rather than type() for a typecheck.
ERROR: plugins/module_utils/controller_api.py:260:23: unidiomatic-typecheck: Use isinstance() rather than type() for a typecheck.
ERROR: plugins/modules/ad_hoc_command.py:166:39: unidiomatic-typecheck: Use isinstance() rather than type() for a typecheck.
ERROR: plugins/modules/import.py:59:4: unused-import: Unused EXPORTABLE_RESOURCES imported from awxkit.api.pages.api
ERROR: test/awx/conftest.py:22:0: unused-import: Unused credentialtype_scm imported from awx.main.tests.functional.conftest
ERROR: test/awx/conftest.py:22:0: unused-import: Unused credentialtype_ssh imported from awx.main.tests.functional.conftest

ERROR: Found 14 validate-modules issue(s) which need to be resolved:
ERROR: plugins/modules/export.py:0:0: nonexistent-parameter-documented: Argument 'applications' is listed in DOCUMENTATION.options, but not accepted by the module argument_spec
ERROR: plugins/modules/export.py:0:0: nonexistent-parameter-documented: Argument 'credential_types' is listed in DOCUMENTATION.options, but not accepted by the module argument_spec
ERROR: plugins/modules/export.py:0:0: nonexistent-parameter-documented: Argument 'credentials' is listed in DOCUMENTATION.options, but not accepted by the module argument_spec
ERROR: plugins/modules/export.py:0:0: nonexistent-parameter-documented: Argument 'execution_environments' is listed in DOCUMENTATION.options, but not accepted by the module argument_spec
ERROR: plugins/modules/export.py:0:0: nonexistent-parameter-documented: Argument 'inventory' is listed in DOCUMENTATION.options, but not accepted by the module argument_spec
ERROR: plugins/modules/export.py:0:0: nonexistent-parameter-documented: Argument 'inventory_sources' is listed in DOCUMENTATION.options, but not accepted by the module argument_spec
ERROR: plugins/modules/export.py:0:0: nonexistent-parameter-documented: Argument 'job_templates' is listed in DOCUMENTATION.options, but not accepted by the module argument_spec
ERROR: plugins/modules/export.py:0:0: nonexistent-parameter-documented: Argument 'notification_templates' is listed in DOCUMENTATION.options, but not accepted by the module argument_spec
ERROR: plugins/modules/export.py:0:0: nonexistent-parameter-documented: Argument 'organizations' is listed in DOCUMENTATION.options, but not accepted by the module argument_spec
ERROR: plugins/modules/export.py:0:0: nonexistent-parameter-documented: Argument 'projects' is listed in DOCUMENTATION.options, but not accepted by the module argument_spec
ERROR: plugins/modules/export.py:0:0: nonexistent-parameter-documented: Argument 'schedules' is listed in DOCUMENTATION.options, but not accepted by the module argument_spec
ERROR: plugins/modules/export.py:0:0: nonexistent-parameter-documented: Argument 'teams' is listed in DOCUMENTATION.options, but not accepted by the module argument_spec
ERROR: plugins/modules/export.py:0:0: nonexistent-parameter-documented: Argument 'users' is listed in DOCUMENTATION.options, but not accepted by the module argument_spec
ERROR: plugins/modules/export.py:0:0: nonexistent-parameter-documented: Argument 'workflow_job_templates' is listed in DOCUMENTATION.options, but not accepted by the module argument_spec
```
<!---
If you are fixing an existing issue, please include "related #nnn" in your
commit message and your description; but you should still explain what
the change does.
-->

##### ISSUE TYPE
<!--- Pick one below and delete the rest: -->
 - Bug, Docs Fix or other nominal change

##### COMPONENT NAME
<!--- Name of the module/plugin/module/task -->
 - Collection

##### AWX VERSION
<!--- Paste verbatim output from `make VERSION` between quotes below -->
```
awx: 24.3.1.dev22+g805e06ed2c.d20240521
```

fosterseth committed:
Fix up ansible sanity checks

gundalow commented on awx_collection/plugins/modules/export.py:
```diff
@@ -137,6 +137,20 @@
 def main():
     argument_spec = dict(
         all=dict(type='bool', default=False),
+        organizations=dict(type='list', elements='str'),
```
This is confusing, how did this work before?

fosterseth replied:
I believe ansible will just inject data from the yaml file, regardless of what is in the argument_spec. But explicitly defining argument_spec allows for better data validation at runtime, so it should be set. That is my understanding anyways

gundalow replied:
`ansible-test` should have been validating this for years, so I'm not sure why it's only failing now

thedoubl3j replied:
@gundalow it has been but I think as of late, ansible has allowed a larger arg_spec to be defined so we might have been ignoring the error (not uncommon).

AlanCoding commented on awx_collection/plugins/modules/import.py:
```diff
@@ -56,8 +56,6 @@
 # In this module we don't use EXPORTABLE_RESOURCES, we just want to validate that our installed awxkit has import/export
 try:
-    from awxkit.api.pages.api import EXPORTABLE_RESOURCES  # noqa
-
```
post-change, this doesn't make any sense.

fosterseth replied:
should we just drop this try except block? do we really need to check whether awxkit has the import/export functions? I think that feature was long enough ago we can assume it is there

AlanCoding replied:
I would not give any resistance to that idea. But since the rest of this merged, maybe not worth it.

TheRealHaoLiu committed:
Fix awx-collection test failure

```
fixture 'credentialtype_ssh' not found
```

TheRealHaoLiu committed:
Fix module adhoc command extra string-ify

example failure
```
<testhost> EXEC /bin/sh -c '/usr/bin/python3 && sleep 0'
fatal: [testhost]: FAILED! => {
    "changed": false,
    "invocation": {
        "module_args": {
            "become_enabled": null,
            "controller_config_file": null,
            "controller_host": null,
            "controller_oauthtoken": null,
            "controller_password": null,
            "controller_username": null,
            "credential": "AWX-Collection-tests-ad_hoc_command-ssh-cred-ThLYQcDoJvqKDIzL",
            "diff_mode": null,
            "execution_environment": null,
            "extra_vars": {
                "var1": "test var"
            },
            "forks": null,
            "interval": 2.0,
            "inventory": "Demo Inventory",
            "job_type": null,
            "limit": null,
            "module_args": null,
            "module_name": "ping",
            "request_timeout": null,
            "timeout": null,
            "validate_certs": null,
            "verbosity": null,
            "wait": true
        }
    },
    "msg": "Failed to launch command, see response for details",
    "response": {
        "json": {
            "extra_vars": [
                "Not a valid string."
            ]
        },
        "status_code": 400
    }
}
```

fosterseth committed:
import EXPORTABLE_RESOURCES

Signed-off-by: Seth Foster <fosterbseth@gmail.com>

TheRealHaoLiu commented:
closing and reopen to trigger CI... github action runners seems to be struggling today?

fosterseth committed:
spaces

Signed-off-by: Seth Foster <fosterbseth@gmail.com>

TheRealHaoLiu commented:
don't merge yet... we gonna try to cary over the 2.16's ignore file and see if that's enough... no clue why they were ignored on 2.16 and previous version tho...https://github.com/ansible/awx/pull/15210

TheRealHaoLiu committed:
Revert export argspec change and add ignore
